edsnlp/utils/training.py

```diff
@@ -0,0 +1,445 @@
+        gold_preproc = false
+        limit = 0
+        augmenter = null
+
+    [corpora.dev]
+        @readers = "spacy.Corpus.v1"
+        path = ${paths.dev}
+        max_length = 0
+        gold_preproc = false
+        limit = 0
+        augmenter = null
+
+[training]
+    train_corpus = "corpora.train"
+    dev_corpus = "corpora.dev"
+    seed = ${system.seed}
+    gpu_allocator = ${system.gpu_allocator}
+    dropout = 0.1
+    accumulate_gradient = 1
+    patience = 10000
+    max_epochs = 0
+    max_steps = 20000
+    eval_frequency = 200
+    frozen_components = []
+    before_to_disk = null
+
+    [training.batcher]
+        @batchers = "spacy.batch_by_words.v1"
+        discard_oversize = false
+        tolerance = 0.2
+        get_length = null
+
+    [training.batcher.size]
+        @schedules = "compounding.v1"
+        start = 100
+        stop = 1000
+        compound = 1.001
+        t = 0.0
+
+    [training.logger]
+        @loggers = "eds.RichLogger.v1"
+        progress_bar = false
+
+    [training.optimizer]
+        @optimizers = "Adam.v1"
+        beta1 = 0.9
+        beta2 = 0.999
+        L2_is_weight_decay = true
+        L2 = 0.01
+        grad_clip = 1.0
+        use_averages = false
+        eps = 0.00000001
+        learn_rate = 0.001
+
+[initialize]
+    vectors = ${paths.vectors}
+    init_tok2vec = ${paths.init_tok2vec}
+    vocab_data = null
+    lookups = null
+    before_init = null
+    after_init = null
+""",
+    interpolate=False,
+)
+
+
+class DataFormat(Enum):
+    brat = "brat"
+    standoff = "brat"
+    spacy = "spacy"
+
+
+def make_spacy_corpus_config(
+    train_data: Union[str, list[Doc]],
+    dev_data: Union[str, list[Doc], int, float],
+    data_format: Optional[DataFormat],
+    nlp: Optional[spacy.Language] = None,
+    seed: int = 0,
+    reader: str = "spacy.Corpus.v1",
+):
+    """
+    Helper to create a spacy's corpus config from training and dev data by
+    loading the documents accordingly and exporting the documents using spacy's DocBin.
+
+    Parameters
+    ----------
+    train_data: Union[str, list[Doc]]
+        The training data. Can be:
+            - a list of spacy.Doc
+            - a path to a given dataset
+    dev_data: Union[str, list[Doc], int, float]
+        The development data. Can be:
+            - a list of spacy.Doc
+            - a path to a given dataset
+            - the number of documents to take from the training data
+            - the fraction of documents to take from the training data
+    data_format: Optional[DataFormat]
+        Optional data format to determine how we should load the documents from the disk
+    nlp: Optional[spacy.Language]
+        Optional spacy model to load documents from non-spacy formats (like brat)
+    seed: int
+        The seed if we need to shuffle the data when splitting the dataset
+    reader: str
+        Which spacy reader to use when loading the data
+
+    Returns
+    -------
+    Config
+    """
```

Comment:
How about this to load the data ? @bdura

Reply:
Looks good!